Signal/src/ViewControllers/DebugUI/DebugUIMessages.m

```diff
@@ -1028,6 +1041,54 @@ + (void)createNewGroups:(int)counter recipientId:(NSString *)recipientId
     [messageSender sendMessage:message success:completion failure:completion];
 }
 
++ (void)injectFakeIncomingMessages:(int)counter thread:(TSThread *)thread
+{
+    // Wait 5 seconds so debug user has time to navigate to another
+    // view before message processing occurs.
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(5.f * NSEC_PER_SEC)),
+        dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0),
+        ^{
+            for (int i = 0; i < counter; i++) {
+                [self injectIncomingMessageInThread:thread counter:counter];
+            }
+        });
+}
+
++ (void)injectIncomingMessageInThread:(TSThread *)thread counter:(int)counter
+{
+    OWSAssert(thread);
+
+    NSString *randomText = [self randomText];
+    NSString *text = [[[@(counter) description] stringByAppendingString:@" "] stringByAppendingString:randomText];
+
+    OWSSignalServiceProtosDataMessageBuilder *dataMessageBuilder = [OWSSignalServiceProtosDataMessageBuilder new];
+    [dataMessageBuilder setBody:text];
+
+    if ([thread isKindOfClass:[TSGroupThread class]]) {
+        TSGroupThread *groupThread = (TSGroupThread *)thread;
+        OWSSignalServiceProtosGroupContextBuilder *groupBuilder = [OWSSignalServiceProtosGroupContextBuilder new];
+        [groupBuilder setType:OWSSignalServiceProtosGroupContextTypeDeliver];
+        [groupBuilder setId:groupThread.groupModel.groupId];
+        [dataMessageBuilder setGroup:groupBuilder.build];
+    }
+
+    OWSSignalServiceProtosContentBuilder *payloadBuilder = [OWSSignalServiceProtosContentBuilder new];
+    [payloadBuilder setDataMessage:dataMessageBuilder.build];
+    NSData *plaintextData = [payloadBuilder build].data;
+
+    OWSSignalServiceProtosEnvelopeBuilder *envelopeBuilder = [OWSSignalServiceProtosEnvelopeBuilder new];
+    [envelopeBuilder setType:OWSSignalServiceProtosEnvelopeTypeCiphertext];
+    [envelopeBuilder setSource:@"+12345678901"];
```

Comment:
Shouldn't this number correspond to the thread rather than hardcoded?

Reply:
Okay I reworked this a bit. Thanks.